sem/utils.py

```diff
@@ -544,19 +544,18 @@ def parse_logs(log_file):
         log_file (string): Path to where the log file is stored
     """
     log_list = []
-    regex = re.compile(r'[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (\[.*?\] )?(\w+):(\w+)\((.*?)\)(: \[(\w+)\s*\] (.*))?')
+    regex = re.compile(r'^[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (?:\[(.*?)\] )?(\w+):(\w+)\((.*?)\)(?:: \[(\w+)\s*\] (.*))?$')
```

Comment:
to improve readibility, we could also think of [named groups](https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups). something like this:
```python
    time_re = r'[\+\-]?(?P<time>\d+\.\d+)s'
    context_re = r'(?P<context>(?:\d+|-\d+))'
    # etc
    regex = re.compile(r'^' + time_re + context_re + ... + r'$')
```
so you can later reference your groups directly, e.g.,
```python
if groups['severity_class'] is None and groups['message'] is None and groups['extended_context'] is None:
   ...
```

NOTE: it's just a suggestion, feel free to avoid adding extra work if you think it unnecessary :)

Reply:
This definitely seems better. Fixed in [8832ada](https://github.com/signetlabdei/sem/pull/51/commits/8832ada1b275d61faf0fba7c6651c683484c5e93).